graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -1417,104 +1451,149 @@ def process_edge_data(self, edge_configs: Sequence[EdgeConfig]) -> Tuple[Dict, D
-                        if edge_config.label_configs[0].task_type != "link_prediction":
-                            self.graph_info["etype_label_property"].append(
-                                edge_config.label_configs[0].label_column
-                            )
-
-                        if self.add_reverse_edges:
-                            # For reverse edges only the label metadata
-                            # (labels + split masks) are relevant.
-                            edge_data_dict[reverse_edge_type] = label_metadata_dicts
-                        self.timers["_process_edge_labels"] += perf_counter() - edge_label_start
-                    else:
-                        # Different edges can have different relation types
-                        # TODO: For each rel_type we need get the label and output it separately.
-                        # We'll create one file output per rel_type, with the labels for each type
-                        raise NotImplementedError(
-                            "Currently we do not support loading edge "
-                            "labels with multiple edge relation types"
+                    edge_type_metadata_dicts.update(label_metadata_dicts)
+                    self.graph_info["etype_label"].append(edge_type)
+                    self.graph_info["etype_label"].append(reverse_edge_type)
+                    self.graph_info["etype_to_label_masks"][edge_type] = []
+
+                    # Collect all the mask names for the edge type
+                    for label_config in edge_config.label_configs:
+                        if label_config.mask_field_names:
+                            mask_names = label_config.mask_field_names
+                        else:
+                            mask_names = ("train_mask", "val_mask", "test_mask")
+                        self.graph_info["etype_to_label_masks"][edge_type].extend(mask_names)
+
+                    if edge_config.label_configs[0].task_type != "link_prediction":
+                        self.graph_info["etype_label_property"].append(
+                            edge_config.label_configs[0].label_column
                         )
-                if edge_type_metadata_dicts:
-                    edge_data_dict[edge_type] = edge_type_metadata_dicts
-                edges_df.unpersist()
 
-        logging.info("Finished processing edge features")
-        return edge_data_dict, edges_dict
+                    if self.add_reverse_edges:
+                        # For reverse edges only the label metadata
+                        # (labels + split masks) are relevant.
+                        edge_data_dict[reverse_edge_type] = label_metadata_dicts
+                    self.timers["_process_edge_labels"] += perf_counter() - edge_label_start
+                else:
+                    # Different edges can have different relation types
+                    # TODO: For each rel_type we need get the label and output it separately.
```

Comment:
Why we need this TODO? The function is supposed to handle just one edge type. So it is impossible that there are multiple relation types.

Reply:
This is a leftover from NeptuneML, where we had the `relname/prefixname` distinction and was planning to support that. We can remove this in a future PR.